### ISSUE TITLE
Better handling of non-existent node_modules directory

### ISSUE THREAD

**Issue by sindresorhus** [benmosher/eslint-plugin-import#271]
I ran this plugin in a project without a `node_modules` directory. I hadn't yet run `npm install`.

This is what I got:

```
test.js:2
✖  2:18  Unexpected use of file extension "" for "ava"         import/extensions
✖  2:18  Unable to resolve path to module 'ava'.               import/no-unresolved
✖  3:23  Unexpected use of file extension "" for "read-chunk"  import/extensions
✖  3:23  Unable to resolve path to module 'read-chunk'.        import/no-unresolved
```

I don't think the `import/extensions` rule should report in this case. It's just noise.

**Comment by benmosher** [comment]
Ah, yeah, seems like package refs (even internal ones) should should not trigger `extensions` (`always` mode).

@vvo: did you imagine the `always` mode would disallow package references? I would think that ought to be a separate rule, or at least an option.

i.e.

```js
import foo from './foo.js' // valid
import React from 'react' // valid if points to a package and not `react.js`

import bar from './bar' // invalid if filename is actually `bar.js`
```

**Comment by vvo** [comment]
Dunno about the "always mode" but your example is valid and expected.

**Comment by vvo** [comment]
But

```js
import bar from './bar' // invalid if filename is actually `bar.js`
```

Actually if there's a folder named "bar" I would want to disable requiring it. Requiring folders involves magic and hides the filetree from the code.

**Comment by benmosher** [comment]
Fair enough. So: a flag (allow-packages) that allows a folder reference?

**Comment by benmosher** [comment]
Actually I just realized that the main issue may be that the extensions rule needs to ignore unresolved imports. 

This is a standard behavior that would resolve @sindresorhus's original issue regardless of what mode he put it in.

**Comment by benmosher** [comment]
fixed by #296
